src/hevm/src/EVM/Expr.hs

```diff
@@ -153,9 +153,8 @@ shr = op2 SHR (\x y -> if x > 256 then 0 else shiftR y (fromIntegral x))
 
 sar :: Expr EWord -> Expr EWord -> Expr EWord
 sar = op2 SAR (\x y ->
-  let msb = testBit y 255
-      shifted = if x > 256 then 0 else shiftR y (fromIntegral x)
-  in if msb then setBit shifted 255 else shifted)
+  let asSigned = (fromIntegral y) :: Int256
+  in fromIntegral $ shiftR asSigned (fromIntegral x))
```

Comment:
The `shiftR` from `Data.Bits` does a logical shift on unsigned types and an arithmetic shift on signed types, so I just convert `y` to be an `Int256` instead of a `Word256` and use the SAR implementation built in to the `Data.Bits` package.

Reply:
Ah, I see. Sorry.